OpenRA.Game/Widgets/WidgetUtils.cs

```diff
@@ -248,6 +248,12 @@ public static string TruncateText(string text, int width, SpriteFont font)
 
 			return trimmed;
 		}
+
+		public static Color GetContrastColor(Color fgColor, Color bgDark, Color bgLight)
+		{
+			var fg = new HSLColor(fgColor);
+			return fg.RGB == Color.White || fg.L > 128 ? bgDark : bgLight;
```

Comment:
The shadows/contrast look *much* better if you change this 128 to 64.  This makes it err towards the dark background for the mid-range colors where the light looks bad but dark is perfectly fine.